catroid/src/main/java/org/catrobat/catroid/utils/Utils.java

```diff
@@ -1104,4 +1104,14 @@ public static boolean isUserLoggedIn(Context context) {
 	public static String getNumberStringForBricks(float value) {
 		return (int) value == value ? "" + (int) value : "" + value;
 	}
+
+	public static <T extends Object> List<T> distinctListByClassOfObjects(List<T> listToDistinct) {
+		List<T> distinctList = new ArrayList<>();
+		HashMap<Class, T> uniqueMap = new HashMap<>();
+		for (T objectInstance : listToDistinct) {
+			uniqueMap.put(objectInstance.getClass(), objectInstance);
+		}
+		distinctList.addAll(uniqueMap.values());
+		return distinctList;
```

Comment:
Perfect! :+1: 


Just a few things. I'm a little picky here :grimacing:


No need to `extends Object`.
Please use `Map` as declaration type.
```java
return new ArrayList<>(uniqueMap.values());
```
The method is a perfect text-book example for tests. :wink: